src/compojure/core.clj

```diff
@@ -102,6 +102,12 @@
       ((mw handler) request)
       (handler request))))
 
+(defn- add-route-information
+  "Returns the request with the matched route assoc'd onto it as a
+  vector of request method and the matched route."
+  [request path]
+  (assoc request :route [(:request-method request) (:source path)]))
```

Comment:
Also, you can't use `:request-method`, as it doesn't necessarily reflect the route method. Instead, pass `method` as an argument in.